src/platform/nrfconnect/FactoryDataProvider.h

```diff
@@ -30,6 +31,38 @@ extern "C" {
 namespace chip {
 namespace DeviceLayer {
 
+struct InternalFlashFactoryData
+{
+    CHIP_ERROR GetFactoryDataPartition(uint8_t ** data, size_t & dataSize)
+    {
+        *data    = reinterpret_cast<uint8_t *>(reinterpret_cast<uint32_t *>(PM_FACTORY_DATA_ADDRESS));
```

Comment:
Can we cast directly to `uint8_t*`? Also, `data` is passed by pointer and `dataSize` by reference. Would be good to make it consistent (e.g. pass `data` as `uint8_t*&`).